File: Backend/backend/src/main/java/com/paint/backend/Shapes/Ellipse.java
package com.paint.backend.Shapes;

import org.json.JSONObject;



public class Ellipse extends Shape {
    private float radiusX, radiusY;

    public Ellipse() {}
    @Override
    public JSONObject draw() {
        JSONObject attrs = new JSONObject().put("id",id).put("x",x).put("y",y).put("stroke",stroke).put("draggable",draggable)
                .put("strokeScaleEnabled",strokeScaleEnabled).put("strokeWidth",strokeWidth)
                .put("fill",fill).put("radiusX",radiusX).put("radiusY",radiusY)
                .put("scaleX", scaleX ).put("scaleY",scaleY);
        return new JSONObject().put("className",className).put("attrs",attrs);
    }
}

File: frontend/src/app/Components/toolbar/shapeskit/shapes/Ellipse.ts
import Konva from "konva";
import { Shape } from "./Shape";

export class Ellipse extends Shape {
    private _radiusX?: number;
    private _radiusY?: number;

    constructor(x?:number, y?:number, radiusX?:number, radiusY?:number, fill?: string, stroke?: string, strokeWidth?: number, draggable?: boolean, scaleX?: number, scaleY?: number){
        super(x, y, fill, stroke, strokeWidth, draggable, scaleX, scaleY);
        this._radiusX = radiusX;
        this._radiusY = radiusY;
    }

    public get radius(): number | undefined {
        return this._radiusX;
    }
    public set radius(value: number | undefined) {
        this._radiusX = value;
    }

    public get radiusY(): number | undefined {
        return this._radiusY;
    }
    public set radiusY(value: number | undefined) {
        this._radiusY = value;
    }

    override getKonva() {
        return new Konva.Ellipse({
            name: 'shape',
            x: this._x,
            y: this._y,
            radiusX: this._radiusX!,
            radiusY: this._radiusY!,
            fill: this._fill,
            stroke: this._stroke,
            strokeWidth: this._strokeWidth,
            draggable: this._draggable,
            scaleX: this._scaleX,
            scaleY: this._scaleY,
            strokeScaleEnabled: false,
        });
    }
}